Add an event editer

DHaytham opened the pull request:
- editEvent page hinzugefuegt um den User zu erlauben die erstellte events zu bearbeiten,

DHaytham committed:
Add an event editer

CeloKr commented:
kannst du bitte noch eine beschreibung hinzufügen, damit wir wissen was es ist. Am besten noch ein Isue dazu verbinden

DHaytham commented:
> kannst du bitte noch eine beschreibung hinzufügen, damit wir wissen was es ist. Am besten noch ein Isue dazu verbinden

ist jetzt erledigt :)

Mac10goesBRRRT commented on app/[instanz]/[organisation]/[user]/editEvent/page.js:
```diff
@@ -0,0 +1,209 @@
+    <StyledPaper>
+      <Box>
+        <Card sx={{ marginBottom: 4, position: 'relative' }}>
+          <Box
+            sx={{
+              position: 'relative',
+              height: 300,
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'center',
+              backgroundImage: `url(${eventData.image})`,
+              backgroundSize: 'cover',
+              backgroundPosition: 'center',
+            }}
```
Können wir hier nicht die styledComponents verwenden?

Mac10goesBRRRT replied:
@CeloKr Schau mal, ob du hier was entsprechendes gebaut hast.

Mac10goesBRRRT commented on app/[instanz]/[organisation]/[user]/preview/page.js:
```diff
@@ -111,7 +131,7 @@ function EventDetails() {
                 </CardContent>
                 <CardActions sx={{ display: 'flex', justifyContent: 'center' }}>
-                  <GreenButton>
+                  <GreenButton onClick={handleEditEvent}>
                     Bearbeiten
```
Der Link führt zu einem 404

DHaytham replied:
done soll jz funktionieren.

Mac10goesBRRRT replied:
hast du noch mal commits gepusht? Seh bei mir noch nichts

DHaytham replied:
musste noch kleinigkeiten bearbeiten, jz kannst du dir es ansehen.

Mac10goesBRRRT commented on app/[instanz]/[organisation]/[user]/preview/page.js:
Ich würde den Teilnehmen Button mindestens disablen, wenn nicht sogar ganz ausblenden.

Der Foto-Upload-Button kann in der Preview auch raus.

Die Preview mal mit einem Deutlich längeren Text testen, schauen wie und wo da die Scrollbar erscheint.

DHaytham replied:
- Habe den Teilnehmer und den Foto-Upload Buttons  ausgeblendet
- Mit längeren Text auf die Preview wird die ganze Seite laenger was meinen meinung nach gut aussieht.

Mac10goesBRRRT replied:
Jo, das schon mal gut!

Mac10goesBRRRT commented on app/[instanz]/[organisation]/[user]/editEvent/page.js:
Der Image Placeholder ist Breit aber nicht hoch. 16:9 Bilder lassen sich so schwerer sinnvoll einfügen. Wie ist hier der Plan? Sollte dem User hier eine Aspect-Ratio empfohlen werden?

DHaytham replied:
soll jz funktionieren .

Mac10goesBRRRT reviewed (changes requested):
Beide Seiten sind meiner Meinung nach evtl besser in _/event/[event-id]/editEvent_ bzw. _/event/[event-id]/preview_ aufgehoben

DHaytham commented:
> Beide Seiten sind meiner Meinung nach evtl besser in _/event/[event-id]/editEvent_ bzw. _/event/[event-id]/preview_ aufgehoben

es gibt noch andere event ordner wie z.B createEvent, myevent und meinEntwurf soll ich dafuer dasselbe machen ?

Mac10goesBRRRT commented:
> > Beide Seiten sind meiner Meinung nach evtl besser in _/event/[event-id]/editEvent_ bzw. _/event/[event-id]/preview_ aufgehoben
> 
> es gibt noch andere event ordner wie z.B createEvent, myevent und meinEntwurf soll ich dafuer dasselbe machen ?

In dem Pullrequest nicht, das kann aber in nem extra Pull-Request gemacht werden. In dem hier nur die Dateien, die schon drin sind.

DHaytham committed:
- remove Teilnehmer and Foto-Upload Buttons.
- fix placeholder so it accepts all aspect ratios
- fix the Bearbeiten button so it redirects to the editEvent page

DHaytham committed:
add more text to test the beschreibung text-area

CeloKr commented:
Meine Meinung: 
myEvent, createEvent etc was man selbst erstellt, soll in der Sturktur bleiben wie wir es haben. Wenn ein Event angelegt wird soll diese unter event/[event_id] angezeigt werden. 

Wie könnte man da drauf? 
z.B. im Terminmanagement auf ein Termin klicken und Termin Details anzeigen, kommt auf auf event/[event_id]

--> Unter myEvent wird man ja nur sehen, welche Veranstaltungen man erstellt hat, diese kann ruhig auf der route /intanz/organisation/user/myEvent bleiben. 

--> wenn ich als Teilnehmer bzw. später auf Meine Teilnahmen klicke, werde ich ja eine Liste an meine Teilnahmen sehen. 
Da könnte man z.b. auch auf Details anzeigen klicken und man kommt auf die Route --> /event/[event_id]. 

so wäre es meiner meinung nach sehr sinnvoll.

Mac10goesBRRRT commented:
> Meine Meinung: myEvent, createEvent etc was man selbst erstellt, soll in der Sturktur bleiben wie wir es haben. Wenn ein Event angelegt wird soll diese unter event/[event_id] angezeigt werden.
> 
> Wie könnte man da drauf? z.B. im Terminmanagement auf ein Termin klicken und Termin Details anzeigen, kommt auf auf event/[event_id]
> 
> --> Unter myEvent wird man ja nur sehen, welche Veranstaltungen man erstellt hat, diese kann ruhig auf der route /intanz/organisation/user/myEvent bleiben.
> 
> --> wenn ich als Teilnehmer bzw. später auf Meine Teilnahmen klicke, werde ich ja eine Liste an meine Teilnahmen sehen. Da könnte man z.b. auch auf Details anzeigen klicken und man kommt auf die Route --> /event/[event_id].
> 
> so wäre es meiner meinung nach sehr sinnvoll.

Also das wäre kein Problem, du machst im backend den Fetch ja sowieso über die id. Dann wäre die ID der ausschlaggebende Punkt, auch weil in einer Orga ja nicht zwingend ich als Mitglied die Veranstaltung erstelle oder editiere.

Myevent würde ich als fetch nutzen um alle meine Events ziehen zu können.

CeloKr commented:
Also meinst du Meine Teilnahmen komplett weg lassen und nur über Meine Veranstaltungen machen?

DHaytham committed:
Merge branch 'development' into Add-event-editer
